domain/src/main/kotlin/no/nav/su/se/bakover/domain/oppdrag/Utbetalingslinje.kt

```diff
@@ -37,57 +38,161 @@ sealed class Utbetalingslinje : KanPlasseresPåTidslinje<Utbetalingslinje> {
+            override val fraOgMed: LocalDate,
+            override val tilOgMed: LocalDate,
+            override var forrigeUtbetalingslinjeId: UUID30?,
+            override val beløp: Int,
+            override val virkningstidspunkt: LocalDate,
+        ) : Endring() {
+            override val linjeStatus = LinjeStatus.REAKTIVERING
+
+            @JsonIgnore
+            override val periode = Periode.create(fraOgMed, tilOgMed)
+
+            constructor(
+                utbetalingslinje: Utbetalingslinje,
+                virkningstidspunkt: LocalDate,
+                clock: Clock,
+            ) : this(
+                id = utbetalingslinje.id,
+                opprettet = Tidspunkt.now(clock),
+                fraOgMed = utbetalingslinje.fraOgMed,
+                tilOgMed = utbetalingslinje.tilOgMed,
+                forrigeUtbetalingslinjeId = utbetalingslinje.forrigeUtbetalingslinjeId,
+                beløp = utbetalingslinje.beløp,
+                virkningstidspunkt = virkningstidspunkt,
+            )
+        }
+
+        enum class LinjeStatus {
+            OPPHØR,
+            STANS,
+            REAKTIVERING;
+        }
+    }
+}
+
+sealed class UtbetalingslinjePåTidslinje : KanPlasseresPåTidslinje<UtbetalingslinjePåTidslinje> {
+    abstract override val opprettet: Tidspunkt
+    abstract override val periode: Periode
+    abstract val beløp: Int
 
+    data class Ny(
+        override val opprettet: Tidspunkt,
+        override val periode: Periode,
+        override val beløp: Int,
+    ) : UtbetalingslinjePåTidslinje() {
         override fun copy(args: CopyArgs.Tidslinje) = when (args) {
             is CopyArgs.Tidslinje.Full -> this.copy()
             is CopyArgs.Tidslinje.NyPeriode -> this.copy(
-                fraOgMed = args.periode.fraOgMed,
-                tilOgMed = args.periode.tilOgMed,
+                periode = args.periode,
             )
         }
     }
 
-    data class Endring constructor(
-        override val id: UUID30,
+    data class Stans(
         override val opprettet: Tidspunkt,
-        override val fraOgMed: LocalDate,
-        override val tilOgMed: LocalDate,
-        override var forrigeUtbetalingslinjeId: UUID30?,
-        override val beløp: Int,
-        val statusendring: Statusendring,
-    ) : Utbetalingslinje() {
-
-        @JsonIgnore
-        override val periode = Periode.create(fraOgMed, tilOgMed)
-
-        constructor(
-            utbetalingslinje: Utbetalingslinje,
-            statusendring: Statusendring,
-        ) : this(
-            id = utbetalingslinje.id,
-            opprettet = Tidspunkt.now(),
-            fraOgMed = utbetalingslinje.fraOgMed,
-            tilOgMed = utbetalingslinje.tilOgMed,
-            forrigeUtbetalingslinjeId = utbetalingslinje.forrigeUtbetalingslinjeId,
-            beløp = utbetalingslinje.beløp,
-            statusendring = statusendring,
-        )
-
+        override val periode: Periode,
+        override val beløp: Int = 0,
```

Comment:
Denne bør kanskje ikke defaulte til 0 lenger? Skal ikke den tilsvare forrige linje sitt beløp?

Reply:
Dette representerer bare "visningen" av et opphør på tidslinja og er ikke nøyaktig lik det vi bruker i vår kommunikasjon med oppdrag. Frem til nå har vi i hvert fall representert opphør og stans som perioder med beløp = 0 (som vel for så vidt stemmer ganske bra). Beløpet her er også relevant i forholdt til sjekk av f.eks 10%, i så måte _kan_ det også gi mening i den forstand - litt avhengig av hvordan vi ønsker å forholde oss til sjekk av 10% for opphørt periode.